From 6640832e98f886c55178acefe4aaa246960c574c Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:52 -0400
Subject: [ethernet] qlcnic: Initialize mailbox cmd structure to zero

Message-id: <1408626422-3941-129-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88650
O-Subject: [PATCH RHEL 7.1 128/138] qlcnic: Initialize mailbox cmd structure to zero
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Rajesh Borundia <rajesh.borundia@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id c531692073847fe64635f6992d774fb319e3529a

o Uninitialzed fields in mailbox command structure
  caused commands to time out randomly due to garbage
  values so initialize it to zero.

Signed-off-by: Rajesh Borundia <rajesh.borundia@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c
index 498fa63..0a6018c 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c
@@ -457,6 +457,7 @@ static int qlcnic_sriov_get_vf_acl(struct qlcnic_adapter *adapter)
  struct qlcnic_cmd_args cmd;
  int ret = 0;
 
+ memset(&cmd, 0, sizeof(cmd));
  ret = qlcnic_sriov_alloc_bc_mbx_args(&cmd, QLCNIC_BC_CMD_GET_ACL);
  if (ret)
   return ret;
@@ -1462,6 +1463,7 @@ static int qlcnic_sriov_channel_cfg_cmd(struct qlcnic_adapter *adapter, u8 cmd_o
  struct qlcnic_vf_info *vf = &adapter->ahw->sriov->vf_info[0];
  int ret;
 
+ memset(&cmd, 0, sizeof(cmd));
  if (qlcnic_sriov_alloc_bc_mbx_args(&cmd, cmd_op))
   return -ENOMEM;
 
@@ -2032,6 +2034,7 @@ int qlcnic_sriov_cfg_vf_guest_vlan(struct qlcnic_adapter *adapter,
  struct qlcnic_cmd_args cmd;
  int ret;
 
+ memset(&cmd, 0, sizeof(cmd));
  if (vid == 0)
   return 0;
 
-- 
1.7.1